src/display.c

```diff
@@ -566,15 +579,32 @@ set_terminal_modes(void)
 	leaveok(stdscr, false);
 }
 
+static void
+init_tty(void)
+{
+	/* open */
+	opt_tty.file = fopen("/dev/tty", "r+");
+	if (!opt_tty.file)
+		die("Failed to open tty for input");
+	opt_tty.fd = fileno(opt_tty.file);
+
+	/* attributes */
+	opt_tty.attr = calloc(1, sizeof(struct termios));
+	if (!opt_tty.attr)
+		die("Failed allocation for tty attributes");
+	tcgetattr(opt_tty.fd, opt_tty.attr);
+}
+
 void
 init_display(void)
 {
 	bool no_display = !!getenv("TIG_NO_DISPLAY");
 	const char *term;
 	int x, y;
 
+	init_tty();
```

Comment:
This will open `/dev/tty` for tests. I think it would be better to skip this if `no_display` is true.

Reply:
Intentionally so — from the point of view of the OS, a TTY is much more than a display.

The TTY is also the medium for the line discipline, in rough terms the interactive input.

A TTY is also inseparable from process-group leadership via `tcsetpgrp()`, which among other things determines the disposition of signals, which is relevant under the test harness.

Also, regardless of whether a display is visible, tig is still linked with ncurses, and ncurses maintains a thoroughly TTY-oriented understanding of the world in which it runs.   The goal I was driving for here is [for ncurses to keep an input-only TTY under `no_display`](https://github.com/jonas/tig/pull/736/commits/55e84f0488c4b4510ec9ebc6dd923205cafecced).  Tig still does not *display* anything, but the input "side" of ncurses remains connected to a proper [`ioctl`able TTY](https://github.com/jonas/tig/pull/736/commits/e2a2910cef2bf00d8a99789c8ab4bc2a117fc960#diff-2a68c783d9ab596cb14680c33e4cf5acR421).

The separate matter of process-group leadership becomes relevant when there are long-lived child processes as proposed in #301/#734.  The simplest and most robust way to handle child cleanup is to
 * be process-group leader
 * [`SIGHUP` your own process group at exit](https://github.com/jonas/tig/pull/734/files#diff-28d128864ec9d27d6bdea2ab4c987955R717).

In the most ordinary case, the interactive shell makes tig the process-group leader transparently, and we don't have to think about it.  However, that magic does not happen in non-interactive cases, such as when driven by `tig-pick` or the test harness (where `no_display` happens to be set).  So it is a good idea, when you know you may be invoked from a script, to grab a TTY as a [step toward becoming process-group leader](https://github.com/jonas/tig/pull/736/commits/e822119435b2b7315ec640e2fd9d13ff51c5c969).

Congrats on the release!